### ISSUE TITLE
Valena's Shoulder Extruding When Animated

### ISSUE THREAD

**Issue by amzn-tommy** [o3de/o3de#4644]
**Describe the bug**
There is an issue with Valena's animations where her right shoulder is extruding from the rest of the body. It looks like it could be a skeleton import issue.

**Steps to reproduce**
1. Either use the AtomTest project, or sync it and copy the AtomTest/Valena folder into your project https://github.com/o3de/o3de-atomtest 
2. Create a level in the O3DE editor
3. Create an entity with an Actor component and a Simple Motion component
4. Choose Valena.actor as the actor asset on the actor component
5. Enable the 'Draw skeleton' option in the actor component
5. Choose Idle.motion as the motion asset on the simple motion component
6. Enable 'Preview in Editor' on the simple motion component

**Expected behavior**
Valena animates and the body stays intact

**Actual behavior**
There are some strange looking bones in the skeleton, and the right shoulder is disjointed from the rest of the body

**Assets required**
Valena folder from https://github.com/o3de/o3de-atomtest

**Screenshots/Video**
Busted shoulder circled in blue, strange bone circled in red
![ValenaBusted](https://user-images.githubusercontent.com/82672795/137041298-0f3444d2-2982-4240-a3ca-c6c819fda6f8.PNG)

**Found in Branch**
development branch commit 5f52664026774c549d1fefbf4141be884873b8d4 2021-10-07 13:04:10

It also reproduces as far back as 05b75ea33c4b9a3b477b43186a315f7cfeffdc63 2021-09-23 13:04:09

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Windows
 - Version: 10
 - CPU: Intel I9-10900X
 - GPU: NVidia GeForce RTX 2080 SUPER
 - Memory: 64GB

**Additional context**

**Comment by amzn-tommy** [comment]
I've narrowed this down a bit. inside MeshBuilderSkinningInfo::Optimize I'm seeing that we have a bunch of duplicate influences. We then sort the influences by weight, and remove all but the first N=maxNumWeightsPerVertex instances. But that leaves us with duplicates of the highest weighted influence, which each refer to the same jointId. So we're effectively stripping the influences down to 1, at least in the case of the simplified test asset I have created for https://github.com/o3de/o3de/issues/5690. I will have to re-test this theory with Valena to see if the same thing may be happening there, or if that is a different issue.
